web3/middleware/pythonic.py

```diff
@@ -265,6 +284,7 @@ def to_hexbytes(num_bytes, val, variable_length=False):
         ),
         'eth_getCode': apply_formatter_at_index(block_number_formatter, 1),
         'eth_getStorageAt': apply_formatter_at_index(block_number_formatter, 2),
+        'eth_getProof': apply_formatter_at_index(block_number_formatter, 2),
```

Comment:
It's tricky to navigate all the formatters, especially split ones like this and the `rpc_abi` ones. Nice work tracking all these down!